### PR TITLE
[New Rule] AWS Credentials Searched For Inside a Container

### PR DESCRIPTION
**Pull request by imays11**
## Issues

https://github.com/elastic/detection-rules/issues/2881
## Summary
This rule detects AWS credentials searched for inside a container

updated query Screenshot

image

### CODE REVIEW

File: rules/integrations/cloud_defend/credential_access_aws_creds_search_inside_a_container.toml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  event.type== "start" and
+ (
+/*account for tools that execute utilities as a subprocess, in this case the target utility name will appear as a process arg*/
+    ((process.name : ("grep", "egrep", "fgrep") or process.args: ("grep", "egrep", "fgrep")) and
+        process.args : ("*aws_access_key_id*", "*aws_secret_access_key*", "*aws_session_token*", "*accesskeyid*", "*secretaccesskey*")) or 
+
+/*account for tools that execute utilities as a subprocess, in this case the target utility name will appear as a process arg*/
+     ((process.name: ("find", "locate", "mlocate") or process.args: ("find", "locate", "mlocate")) and
+      process.args : "*.aws/credentials")
+  )
```
[the remark this answers is not in the document]
Comment (imays11) [reply rules/integrations/cloud_defend/credential_access_aws_creds_search_inside_a_container.toml:39]: In my mind you might use grep to look for those args after doing something like `env` and might use find to actually locate the `.aws/credentials` file. But I may be limiting the use case too much by thinking that way so I'm not opposed to expanding it the way you've done.

[the remark this answers is not in the document]
Comment (imays11) [reply rules/integrations/cloud_defend/credential_access_aws_creds_search_inside_a_container.toml]: @Aegrah I updated the query

File: rules/integrations/cloud_defend/credential_access_aws_creds_search_inside_a_container.toml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = """
+process where event.module== "cloud_defend" and     
+  event.type== "start" and
+  
+/*account for tools that execute utilities as a subprocess, in this case the target utility name will appear as a process arg*/
+(process.name : ("grep", "egrep", "fgrep", "find", "locate", "mlocate") or process.args: ("grep", "egrep", "fgrep", "find", "locate", "mlocate")) and 
+process.args : ("*aws_access_key_id*", "*aws_secret_access_key*", "*aws_session_token*", "*accesskeyid*", "*secretaccesskey*", "*access_key*", "*.aws/credentials*")
```
Comment (Aegrah) [inline rules/integrations/cloud_defend/credential_access_aws_creds_search_inside_a_container.toml:34]:
```suggestion
process where event.module == "cloud_defend" and     
  event.type == "start" and
  
/*account for tools that execute utilities as a subprocess, in this case the target utility name will appear as a process arg*/
(process.name : ("grep", "egrep", "fgrep", "find", "locate", "mlocate") or process.args : ("grep", "egrep", "fgrep", "find", "locate", "mlocate")) and 
process.args : ("*aws_access_key_id*", "*aws_secret_access_key*", "*aws_session_token*", "*accesskeyid*", "*secretaccesskey*", "*access_key*", "*.aws/credentials*")
```

**Review by Aegrah — APPROVED** [review]
Just a final nit for consistency. LGTM!

**Review by DefSecSentinel — APPROVED** [review]
LGTM

**Pull request merged**